// @lighthouse-web3/sdk/dist/Lighthouse/upload/text/node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const lighthouse_config_1 = require("../../../lighthouse.config");
const util_1 = require("../../utils/util");
exports.default = async (text, apiKey, name) => {
    try {
        const token = 'Bearer ' + apiKey;
        const endpoint = lighthouse_config_1.lighthouseConfig.lighthouseNode + '/api/v0/add';
        // Upload file
        const formData = new FormData();
        const blob = new Blob([Buffer.from(text)]);
        formData.append('file', blob, name);
        const response = await (0, util_1.fetchWithTimeout)(endpoint, {
            method: 'POST',
            body: formData,
            credentials: 'include',
            timeout: 7200000,
            headers: {
                'Mime-Type': 'text/plain',
                Authorization: token,
            },
        });
        if (!response.ok) {
            throw new Error(`Request failed with status code ${response.status}`);
        }
        const data = await response.json();
        return { data };
    }
    catch (error) {
        throw new Error(error?.message);
    }
};
